### PR TITLE
rgw_crypt: log error messages during failures

### PR DESCRIPTION
**Pull request by theanalyst**
Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

### CODE REVIEW

File: src/rgw/rgw_crypt.cc [hunk 1]
```diff
@@ -1022,15 +1022,20 @@ int rgw_s3_prepare_encrypt(struct req_state* s,
         get_crypt_attribute(s->info.env, parts, X_AMZ_SERVER_SIDE_ENCRYPTION_CUSTOMER_ALGORITHM);
     if (! req_sse_ca.empty()) {
       if (req_sse_ca != "AES256") {
+        ldout(s->cct, 5) << "ERROR: Invalid value for header"
+                         << "x-amz-server-side-encryption-customer-algorithm"
```
Comment (cbodley) [inline src/rgw/rgw_crypt.cc]: missing space between header and x-amz

Comment (theanalyst) [inline src/rgw/rgw_crypt.cc]: ack, just had a  line break to fill column < 80, will fix

Comment (aclamk) [inline src/rgw/rgw_crypt.cc]: @theanalyst I like the debug messages you introduced. I think encryption failures in L1170 L1180 L1194 should also be noted.

**Comment by theanalyst** [comment]
@aclamk @cbodley @joscollin thanks for the reveiw, updated.

**Comment by theanalyst** [comment]
jenkins test this please

**Comment by cbodley** [comment]
jenkins test this please

**Comment by joscollin** [comment]
jenkins test this please

**Comment by yuriw** [comment]
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-08-04_22:48:29-rgw-wip-yuri-testing3_2017_8_5-distro-basic-smithi/

rerun of failed:

http://pulpito.ceph.com/yuriw-2017-08-05_16:08:03-rgw-wip-yuri-testing3_2017_8_5-distro-basic-smithi/

**Pull request merged**